moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -148,10 +145,17 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
     }
     else
     {
-      std::future_status status = result_future.wait_for(timeout.to_chrono<std::chrono::seconds>());
+      std::future_status status = result_future.wait_for(timeout.to_chrono<std::chrono::duration<double>>());
       if (status == std::future_status::timeout)
+      {
         RCLCPP_WARN(LOGGER, "waitForExecution timed out");
+        return false;
+      }
     }
+    // workaround for controllerDoneCallback being called a little bit late
+    rclcpp::Time deadline = node_->now() + rclcpp::Duration::from_seconds(0.1);  // limit waiting to 0.1s
```

Comment:
Should this be a constant (same with the 100ns below)?

Reply:
Please don't add constant delays. Better use a way to explicitly wait for the done callback (i.e. condition variable, promise).